### PR TITLE
[Manta-PC] Updrading dependencies to `v0.9.10`

### LINKED ISSUE #196: Upgrade dependencies to `v0.9.10`
polkadot -> `release-v0.9.10`
substrate -> `polkadot-v0.9.10`
cumulus -> `polkadot-v0.9.10`

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes: #196 

Polkadot Changes:

* paritytech/polkadot#3763
* https://github.com/paritytech/polkadot/pull/3865
* paritytech/polkadot#3672
* paritytech/polkadot#3676
* paritytech/polkadot#3684
* paritytech/polkadot#3761
Overview PR:
* https://github.com/paritytech/polkadot/issues/3760

Substrate Changes:
* https://github.com/paritytech/substrate/pull/9692
* https://github.com/paritytech/substrate/pull/9715
* https://github.com/paritytech/substrate/pull/9704
* https://github.com/paritytech/substrate/pull/9767
* https://github.com/paritytech/substrate/pull/9845
* No overview PR

Cumulus Changes:
* https://github.com/paritytech/cumulus/pull/561
* https://github.com/paritytech/cumulus/pull/570
* https://github.com/paritytech/cumulus/commit/fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193
* No overview PR

`statemine` diffs between:
* `fd80849dde5c209c20a996cfcc5aaacd4666dcbe` -> `0.9.9`
* `fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193` -> `0.9.10`

Performed tests with local `calamari-local` and `calamari-testnet` networks:

- [x] Does key injection script work (only for calamari-testnet)
- [x] Check block production
- [x] Sudo `system.setCode` upgrade
- [x] Force transfer balances
- [x] Pass sudo to multi-sig account
- [x] Upgrade with `authorizeUpgrade + enactAuthorized`
- [x] Check average block time is ~12 seconds
- [x] Check dropping and reconnecting a collator

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (`manta` or `manta-pc`) with right title (start with [Manta] or [Manta-PC]),
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [x] Updated relevant documentation in the code.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. If this number is updated, then the spec_version must also be updated 
- [x] If needed, notify the committer this is a draft-release and a tag is needed after merging the PR.

### CODE REVIEW

File: runtime/manta-pc/src/lib.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub use sp_runtime::BuildStorage;
 use pallet_xcm::{EnsureXcm, IsMajorityOfBody, XcmPassthrough};
 use polkadot_parachain::primitives::Sibling;
 use polkadot_runtime_common::{BlockHashCount, RocksDbWeight, SlowAdjustingFeeUpdate};
-use xcm::v0::{BodyId, Junction, MultiLocation, NetworkId};
+use xcm::latest::prelude::*;
```
Comment (stechu) [inline runtime/manta-pc/src/lib.rs:51]: Can we find a way to avoid using `latest`?
Also, can we not use `*`?

Comment (ghzlatarev) [reply runtime/manta-pc/src/lib.rs:51]: This seems to be fine actually. Here's all that is included:
https://github.com/paritytech/polkadot/blob/release-v0.9.10/xcm/src/v1/mod.rs#L52-L81
And these modules are all private so i can't include them directly.

**Comment by ghzlatarev** [comment]
@stechu reverted the `chain_spec` version to 1. We will need a runtime upgrade after Kusama has done one, as per what Basti said.

**Comment by ghzlatarev** [comment]
@stechu triggered the integration tests with a fake tag name`v0.0.0-polkadot-v0.9.9-tests` 
We will need a real one when you merge this.

**Comment by stechu** [comment]
> @stechu reverted the `chain_spec` version to 1. We will need a runtime upgrade after Kusama has done one, as per what Basti said.

I am not sure whether this alone will do that job. In the code, you still rely on the new version of XCM. I would suggest you to roll-back the XCM related code, i.e. still use `v0`.

**Comment by stechu** [comment]
@ghzlatarev  As we discussed, can you provide the list of upstream changes that we can review against to?

**Comment by ghzlatarev** [comment]
> @ghzlatarev As we discussed, can you provide the list of upstream changes that we can review against to?

My bad included it in the description.
polkadot -> release-v0.9.10 -> https://github.com/paritytech/polkadot/releases/tag/v0.9.10-rc2
Cumulus diff between:
fd80849dde5c209c20a996cfcc5aaacd4666dcbe -> 0.9.9
fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193 -> 0.9.10

**Comment by ghzlatarev** [comment]
> @stechu triggered the integration tests with a fake tag name`v0.0.0-polkadot-v0.9.9-tests`
> We will need a real one when you merge this.

Interesting, the integration tests and runtime upgrade test failed. Have to resolve that.

**Comment by ghzlatarev** [comment]
> > @stechu reverted the `chain_spec` version to 1. We will need a runtime upgrade after Kusama has done one, as per what Basti said.
> 
> I am not sure whether this alone will do that job. In the code, you still rely on the new version of XCM. I would suggest you to roll-back the XCM related code, i.e. still use `v0`.

I don't seem to be able to make the runtime code compile without using `v1` folder yet.

**Comment by Dengjianping** [comment]
@stechu Do you think we need to bump the version as well? From `3.0.1` to `3.0.2`.

**Comment by stechu** [comment]
> @stechu Do you think we need to bump the version as well? From `3.0.1` to `3.0.2`.

We should.

**Comment by stechu** [comment]
> > > @stechu reverted the `chain_spec` version to 1. We will need a runtime upgrade after Kusama has done one, as per what Basti said.
> > 
> > 
> > I am not sure whether this alone will do that job. In the code, you still rely on the new version of XCM. I would suggest you to roll-back the XCM related code, i.e. still use `v0`.
> 
> I don't seem to be able to make the runtime code compile without using `v1` folder yet.

**Comment by stechu** [comment]
> > @ghzlatarev As we discussed, can you provide the list of upstream changes that we can review against to?
> 
> My bad included it in the description.
> polkadot -> release-v0.9.10 -> https://github.com/paritytech/polkadot/releases/tag/v0.9.10-rc2
> Cumulus diff between:
> fd80849dde5c209c20a996cfcc5aaacd4666dcbe -> 0.9.9
> fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193 -> 0.9.10

@ghzlatarev  Here is an example of how to properly description a upstream upgrade:

Polkadot Changes:

The list of polkadot changes
* https://github.com/paritytech/polkadot/pull/3672
* https://github.com/paritytech/polkadot/pull/3676
* https://github.com/paritytech/polkadot/pull/3684
* https://github.com/paritytech/polkadot/pull/3761

Overview PR:
https://github.com/paritytech/polkadot/pull/3763

Please do the same for Cumulus and Substrate.

**Comment by ghzlatarev** [comment]
Substrate Changes:
* https://github.com/paritytech/substrate/pull/9692
* https://github.com/paritytech/substrate/pull/9715
* https://github.com/paritytech/substrate/pull/9704
* https://github.com/paritytech/substrate/pull/9767
* https://github.com/paritytech/substrate/pull/9845
* No overview PR

Cumulus Changes:
* https://github.com/paritytech/cumulus/pull/561
* https://github.com/paritytech/cumulus/pull/570
* https://github.com/paritytech/cumulus/commit/fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193
* No overview PR

`statemine` diffs between:
* `fd80849dde5c209c20a996cfcc5aaacd4666dcbe` -> `0.9.9`
* `fa116cfbfb470a8114fc02f9d2e3c0a5ab6c7193` -> `0.9.10`

**Comment by ghzlatarev** [comment]
> > @stechu Do you think we need to bump the version as well? From `3.0.1` to `3.0.2`.
> 
> We should.

Done.

**Review by Dengjianping — APPROVED** [review]
LGTM!

**Pull request merged**